source/index.ts

```diff
@@ -185,7 +185,18 @@ is.symbol = isOfType<symbol>('symbol');
 is.numericString = (value: unknown): value is string =>
 	is.string(value) && !is.emptyStringOrWhitespace(value) && !Number.isNaN(Number(value));
 
-is.array = Array.isArray;
+is.array = <T = unknown>(value: unknown, assertion?: (value: T) => void): value is T[] => {
```

Comment:
```suggestion
is.array = <T = unknown>(value: unknown, assertion?: (value: T) => value is T): value is T[] => {
```

On the `is` method, we're not asserting anything, and we don't throw exceptions.
Your `assertion` parameter returns `void`, which is not what is expected from `is.array`, and you can actually give the wrong time of method as the `assertion` param and it would allow it (and might return the wrong value).

The `assert.array` methods should in fact work with `asserts ...` in the typing.